Doc/whatsnew/3.14.rst

```diff
@@ -1075,6 +1075,51 @@ turtle
   (Contributed by Marie Roald and Yngve Mardal Moe in :gh:`126350`.)
 
 
+types
+-----
+
+* :class:`types.UnionType` is now an alias for :class:`typing.Union`.
+  See :ref:`below <whatsnew314-typing-union>` for more details.
+  (Contributed by Jelle Zijlstra in :gh:`105499`.)
+
+
+typing
+------
+
+.. _whatsnew314-typing-union:
+
+* :class:`types.UnionType` and :class:`typing.Union` are now aliases for each other,
+  meaning that both old-style unions (created with ``Union[int, str]``) and new-style
+  unions (``int | str``) now create instances of the same runtime type. This unifies
+  the behavior between the two syntaxes, but leads to some differences in behavior that
+  may affect users who introspect types at runtime:
+
+  - Both syntaxes for creating a union now produce the same string representation in
+    ``repr()``. For example, ``repr(Union[int, str])``
+    is now ``"int | str"`` instead of ``"typing.Union[int, str]"``.
+  - Unions created using the old syntax are no longer cached. Previously, running
+    ``Union[int, str]`` multiple times would return the same object
+    (``Union[int, str] is Union[int, str]`` would be ``True``), but now it will
+    return two different objects. Users should use ``==`` to compare unions for equality, not
+    ``is``. New-style unions have never been cached this way.
+    This change could increase memory usage for some programs that use a large number of
+    unions, but unions used in annotations are no longer evaluated by default in Python 3.14
+    (because of :pep:`649`), the new union object is itself much smaller, and removing the
+    cache also saves some space, so it is unlikely that this change will cause a significant
+    increase in memory usage.
```

Comment:
```suggestion
    unions using ``typing.Union`` subscripts. However, this is offset by the fact that
    unions used in annotations are no longer evaluated by default in Python 3.14
    (because of :pep:`649`), the fact that an instance of :class:`types.UnionType` is
    itself much smaller than the object returned by ``Union[]`` was on prior Python
    versions, and the fact that removing the cache also saves some space. it is therefore
    unlikely that this change will cause a significant increase in memory usage for most
    users.
```

Reply:
I don't like the repetition of "the fact that" but I appreciate that my sentence got too long, I'll push another rewrite.

Reply:
yeah I didn't like that aspect of my suggestion either, feel free to tweak it 😆

Reply:
How about this?

Reply:
LGTM!